lib/libuutil/uu_list.c

```diff
@@ -416,8 +416,11 @@ list_walk_init(uu_list_walk_t *wp, uu_list_t *lp, uint32_t flags)
 		 */
 		wp->ulw_next = next = &lp->ul_null_walk;
 		wp->ulw_prev = prev = next->ulw_prev;
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdangling-pointer="
 		next->ulw_prev = wp;
 		prev->ulw_next = wp;
+#pragma GCC diagnostic pop
```

Comment:
We've needed to wrap the pragma like this elsewhere:

```
#if defined(__GNUC__) && !defined(__clang__)
...
#endif
```

Reply:
Alternately, what if you took it off the stack the silence the warning, `uu_zalloc()` / `uu_free()`.  This is a debug path I suspect we've never ever run.  Dropping this core entirely seems fine too.  Just slightly less minimal.

Reply:
My latest push allocates `my_walk` so it's no longer on the stack.